### ISSUE TITLE
Flag to disable browserify build

### ISSUE THREAD

**Issue by user471** [purescript-contrib/pulp#185]
I build a project from the editor so there is no need to also build it with browserify command.

**Comment by hdgarrood** [comment]
If you only want to browserify, why not run browserify directly?

**Comment by user471** [comment]
Could you show me how the command looks like?

**Comment by hdgarrood** [comment]
I think it will be something like `browserify output/*/index.js > bundle.js`. I think you can also do something like `psc-bundle 'output/*/*.js' | browserify > bundle.js`.

Still, the fact that `pulp browserify` always calls `psc` shouldn't affect its running time very much, as `psc` itself takes care of working out if anything needs compiling again.

**Comment by user471** [comment]
> I think it will be something like browserify output/*/index.js > bundle.js

Doesn't work, I think it accepts only  one file. but with `browserify output/main/index.js` it cannot find another modules.

> Still, the fact that pulp browserify always calls psc shouldn't affect its running time very much, as psc itself takes care of working out if anything needs compiling again.

The problem is `pulp browserify --watch`. When I save a file my editor compile the project and simultaneously pulp also compile the project and sometime they can corrupt output js files.

**Comment by hdgarrood** [comment]
Ah, I see. And you want your editor to do the compiling because then it can display errors inline, is that right?

**Comment by user471** [comment]
Yes

**Comment by hdgarrood** [comment]
What if you told your editor to use `pulp browserify` as the compile command?

**Comment by user471** [comment]
It would be slower because of browsefication part.
Also it doesn't support `psc` `--output` option

**Comment by hdgarrood** [comment]
Ok, just to make sure I've understood the problem: can you try watching `output` and, whenever it changes, run this:

```
psc-bundle 'output/*/*.js'  --module=Main --main=Main | browserify > bundle.js
```

**Comment by hdgarrood** [comment]
I think the easiest way to do that would be to use chokhidar-cli: https://github.com/kimmobrunfeldt/chokidar-cli

**Comment by user471** [comment]
Do you want to replace pulp with `chokidar-cli` and `psc-bundle 'output/*/*.js'  --module=Main --main=Main | browserify > bundle.js`?
It's possible, but pulp already has this functionality and it would be better to just add a new flag.

**Comment by hdgarrood** [comment]
No, I just want to understand your problem properly.

**Comment by bodil** [comment]
`pulp browserify` actually used to have a `--skip-compile` option. Guess it got lost in translation - I'd vote for putting it back, it can be really helpful to split the build and browserify steps with larger builds.

**Comment by user471** [comment]
chokidar and psc-pundle works, but there is an error for foreign file

```
// module Main

x = y => 1;
```

```
Error in module Main (Foreign):

  The module could not be parsed:
  GtToken {tokenSpan = TokenPn 23 3 8, tokenComment = [NoComment]}
```

**Comment by hdgarrood** [comment]
ES6 isn't supported in FFI modules; you would see the same error if you were just using `pulp browserify`.

I've just checked and compiling slamdata with psc 0.8.2 takes about 3 seconds if everything is already up-to-date, which is a bit longer than I thought it would be, so :+1: for putting back `--skip-compile`, then.

I don't think `pulp browserify --skip-compile` would work well in @user471's situation, though, because it would be kicked off at the same time as the rebuild would, which would mean that `pulp browserify` would most likely pick up out-of-date versions of compiled JS in `output`. That's why we need to be watching `output` instead.

**Comment by hdgarrood** [comment]
Unless `pulp --watch browserify --skip-compile` were made to watch `output` instead, somehow. :/

**Comment by user471** [comment]
> you would see the same error if you were just using pulp browserify

`pulp browserify` compile this without any errors. I don't think `pulp` use psc-publish

**Comment by bodil** [comment]
Yes, it's only `psc-bundle` specifically which tries to parse the JS and doesn't handle ES6 syntax. If you don't use `psc-bundle` there should be no problem.

**Comment by user471** [comment]
@bodil is there any simple way to use `browsefiry` without `pulp` on `output` folder?

**Comment by bodil** [comment]
The problem is that `browserify` needs to have its basedir configured, and I can't see any command line option for that. Something like `NODE_PATH=./output browserify output/Main/index.js` might work, though that's just me guessing that Browserify checks the NODE_PATH variable.

**Comment by hdgarrood** [comment]
Ah, of course, you weren't using `psc-bundle`. Right. You would see the same error if you tried to use `pulp browserify -O`, then.

If you decide to use ES6 in your FFI modules, you miss out on high-quality DCE that plays extremely well with the rest of the purescript tool chain via `psc-bundle`, so I would really discourage it. Especially so if you're writing code that might ever be released as a library. You could always put your ES6 code outside of the FFI modules and bring it in using `require()`.

I think @bodil's suggestion could work. Additionally, you wouldn't need to modify `NODE_PATH` if you ran `pulp build --require-path="../"` to build (it sounds like this might become the default soon, too).

**Comment by user471** [comment]
Thanks. I ended up with this and it seems to work fine.

```
set NODE_PATH=./output
watchify output/browserify.js -v -o bundle.js
```

**Comment by hdgarrood** [comment]
Closing as this seems resolved (but also see #232).
